src/placeos-frontend-loader/api/repositories.cr

```diff
@@ -54,5 +79,26 @@ module PlaceOS::FrontendLoader::Api
           hash[folder_name] = Compiler::Git.current_repository_commit(folder_name, content_directory)
         }
     end
+
+    # Clean repository copies to query
+    ###########################################################################
+
+    class_property query_directory : String do
+      File.join(Dir.tempdir, "loader-queries").tap(&->Dir.mkdir_p(String))
+    end
+
+    def self.with_query_directory(folder, loader : Loader = Loader.instance)
+      authoritative_path = File.join(loader.content_directory, folder)
+      key = Git.repository_lock(authoritative_path).read do
+        remote = Git.remote(folder, loader.content_directory)
+
+        Digest::SHA1.base64digest(remote)[0..6].tap do |remote_digest|
+          cache_path = File.join(query_directory, remote_digest)
+          FileUtils.cp_r(authoritative_path, cache_path) unless Dir.exists?(cache_path)
```

Comment:
Any associated cleanup of this worth worrying about?

Reply:
could introduce an LRU cache with evictions when it’s over some cache size

Reply:
might be worth doing that for build too

Reply:
I think this isn’t that much of a worry _at the moment_ given the user will be unlikely to have more than a handful of frontends.

The LRU cache will be easy to layer over an exisiting directory.

Still worth putting into an issue.

Reply:
Resolving in favour of https://github.com/PlaceOS/build/issues/4